core/pfe-core/controllers/floating-dom-controller.ts

```diff
@@ -94,33 +102,52 @@ export class FloatingDOMController implements ReactiveController {
     host.addController(this);
     this.#options = options as Required<FloatingDOMControllerOptions>;
     this.#options.invoker ??= host;
-    this.#options.arrow ??= false;
-    this.#options.flip ??= true;
     this.#options.shift ??= true;
   }
 
   hostDisconnected() {
     this.#cleanup?.();
   }
 
-  async #update(placement: Placement = 'top', offset?: Offset) {
-    const { flip, padding, shift } = this.#options;
+  async #update(placement: Placement = 'top', offset?: Offset, flip = true, fallbackPlacements?: Placement[]) {
+    const { padding, shift } = this.#options;
 
     const invoker = this.#invoker;
     const content = this.#content;
+    const arrow = this.#arrow;
     if (!invoker || !content) {
       return;
     }
-    const { x, y, placement: _placement } = await computePosition(invoker, content, {
+    const { x, y, placement: _placement, middlewareData } = await computePosition(invoker, content, {
       strategy: 'absolute',
       placement,
       middleware: [
         offsetMiddleware(offset),
         shift && shiftMiddleware({ padding }),
-        flip && flipMiddleware({ padding }),
+        arrow && arrowMiddleware({ element: arrow, padding: arrow.offsetHeight / 2 }),
+        flip && flipMiddleware({ padding, fallbackPlacements }),
       ].filter(Boolean)
     });
 
+    if (arrow) {
+      const { x: arrowX, y: arrowY } = middlewareData.arrow || {};
+
+      const staticSide = {
+        top: 'bottom',
+        right: 'left',
+        bottom: 'top',
+        left: 'right',
+      }[_placement.split('-')[0]] || '';
+
+      Object.assign(arrow.style, {
+        left: arrowX != null ? `${arrowX}px` : '',
+        top: arrowY != null ? `${arrowY}px` : '',
```

Comment:
Noticed a small issue in here where if you set the position to `left` but don't leave enough room it looks like the arrow goes to the middle of the element.  We could add a check like `top: arrowY != null && ['top', 'bottom'].includes(_placement) ? '${arrowY}px' : '',`